KubuQA.sh

```diff
@@ -28,45 +28,68 @@ check_and_install_tool() {
     fi
 }
 
-# Function to check for a previous Kubuntu Test VM
+# Function to check for a previous Kubuntu Test VM. If not found, create one.
 check_existing_vm(){
+    # Run VBoxManage list vms and capture output
+    vms_output=$(VBoxManage list vms)
+    # Check for "TestKubuntuInstall" Virtual Machine
+    vm_id=$(echo "$vms_output" | grep "\"TestKubuntuInstall\"" | awk '{print $2}' | tr -d '{}')
+    if [ -n "$vm_id" ]; then
+        # Prompt the user with kdialog
+        if kdialog --title "VM Exists" --yesno "The 'TestKubuntuInstall' VM exists (ID: $vm_id). Do you want to keep it?"; then
+            # User chose to keep the VM
+            echo "Keeping 'TestKubuntuInstall' VM."
+            return
+        else
+            # User chose to remove the VM
+            VBoxManage unregistervm "$vm_id" --delete
+            echo "'TestKubuntuInstall' VM has been removed."
+        fi
+    fi
+    # There was no VM or the user chose to remove it
+    VBoxManage createvm --name "TestKubuntuInstall" --register
+    # Create storage controllers for the ISO and VDI
+    VBoxManage storagectl "TestKubuntuInstall" --name "SATA Controller" --add sata --bootable=on
+    VBoxManage storagectl "TestKubuntuInstall" --name "IDE Controller" --add ide --bootable=on
+    echo "A new 'TestKubuntuInstall' VM has been created."
+}
 
-# Run VBoxManage list vms and capture output
-vms_output=$(VBoxManage list vms)
-
-# Check for "TestKubuntuInstall" Virtual Machine
-vm_id=$(echo "$vms_output" | grep "\"TestKubuntuInstall\"" | awk '{print $2}' | tr -d '{}')
-
-if [ -n "$vm_id" ]; then
-    # Prompt the user with kdialog
-    response=$(kdialog --title "VM Exists" --yesno "The 'TestKubuntuInstall' VM exists (ID: $vm_id). Do you want to keep it?")
+# Function to check for existing Virtual Disk Image. If not found, create one.
+function check_existing_vdi() {
+    # Check if there is already a registered VDI in VirtualBox
+    if VBoxManage list hdds | grep --quiet "TestKubuntuInstall"; then
```

Comment:
Instead of checking for the VDI file, we check whether VirtualBox actually knows about the file. This ensures that we can use it as intended.